lib/AST/SwiftNameTranslation.cpp

```diff
@@ -51,6 +52,14 @@ getNameForObjC(const ValueDecl *VD, CustomNamesOnly_t customNamesOnly) {
   return VD->getBaseName().getIdentifier().str();
 }
 
+std::string swift::objc_translation::
+getErrorDomainStringForObjC(const EnumDecl *ED) {
+  std::string buffer = ED->getParentModule()->getNameStr();
+  buffer += ".";
+  buffer += ED->getNameStr();
```

Comment:
This didn't actually change any behavior. Did you mean to use the `@objc` name? Note that not every `@objc` enum *has* a custom `@objc` name, though.

The fact that we *do* print nested enums makes this trickier than it otherwise would be.

Reply:
The first commit refactored the code; [the second changed the behavior](https://github.com/apple/swift/pull/21452/commits/152158ea974df33eff8dd13c904d7c35d30668fb#diff-b62ed7b05c464546f0a07d1e285c96d9R57).

Reply:
Oh, bah, I must have just been looking at the first one. Nevertheless, the new behavior isn't good enough either:

```swift
class Foo {
  @objc enum Error: Swift.Error { case x }
}

class Bar {
  @objc enum Error: Swift.Error { case x }
}
```